lib/src/lds/df1/efdg2.dart

```diff
@@ -19,4 +36,208 @@ class EfDG2 extends DataGroup {
+    }
+
+    return elements;
+  }
+
+  _readBiometricDataBlock(List<DecodedTV> sbh) {
+    var firstBlock = sbh.first;
+    if (firstBlock.tag.value != BIOMETRIC_DATA_BLOCK_TAG &&
+        firstBlock.tag.value != BIOMETRIC_DATA_BLOCK_CONSTRUCTED_TAG) {
+      throw EfParseError(
+          "Invalid object tag=${firstBlock.tag.value.hex()}, expected tag=$BIOMETRIC_DATA_BLOCK_TAG or $BIOMETRIC_DATA_BLOCK_CONSTRUCTED_TAG ");
+    }
+
+    var data = firstBlock.value;
+    if (data[0] != 0x46 &&
+        data[1] != 0x41 &&
+        data[2] != 0x43 &&
+        data[3] != 0x00) {
+      throw EfParseError("Biometric data block is invalid");
+    }
+
+    var offset = 4;
+
+    versionNumber = _extractContent(data, start: offset, end: offset + 4);
+
+    if (versionNumber != 0x30313000) {
+      throw EfParseError("Version of Biometric data is not valid");
+    }
+
+    offset += 4;
+
+    lengthOfRecord = _extractContent(data, start: offset, end: offset + 4);
+    offset += 4;
+
+    numberOfFacialImages =
+        _extractContent(data, start: offset, end: offset + 2);
+    offset += 2;
+
+    facialRecordDataLength =
+        _extractContent(data, start: offset, end: offset + 4);
+    offset += 4;
+
+    nrFeaturePoints = _extractContent(data, start: offset, end: offset + 2);
+    offset += 2;
+
+    gender = _extractContent(data, start: offset, end: offset + 1);
+    offset += 1;
+
+    eyeColor = _extractContent(data, start: offset, end: offset + 1);
+    offset += 1;
+
+    hairColor = _extractContent(data, start: offset, end: offset + 1);
+    offset += 1;
+
+    featureMask = _extractContent(data, start: offset, end: offset + 3);
+    offset += 3;
+
+    expression = _extractContent(data, start: offset, end: offset + 2);
+    offset += 2;
+
+    poseAngle = _extractContent(data, start: offset, end: offset + 3);
+    offset += 3;
+
+    poseAngleUncertainty =
+        _extractContent(data, start: offset, end: offset + 3);
+    offset += 3;
+
+    // Features (not handled). There shouldn't be any but if for some reason there were,
+    // then we are going to skip over them
+    // The Feature block is 8 bytes
+    offset += nrFeaturePoints * 8;
+
+    faceImageType = _extractContent(data, start: offset, end: offset + 1);
+    offset += 1;
+
+    _imageDataType = _extractContent(data, start: offset, end: offset + 1);
+    offset += 1;
+
+    imageWidth = _extractContent(data, start: offset, end: offset + 2);
+    offset += 2;
+
+    imageHeight = _extractContent(data, start: offset, end: offset + 2);
+    offset += 2;
+
+    imageColorSpace = _extractContent(data, start: offset, end: offset + 1);
+    offset += 1;
+
+    sourceType = _extractContent(data, start: offset, end: offset + 1);
+    offset += 1;
+
+    deviceType = _extractContent(data, start: offset, end: offset + 2);
+    offset += 2;
+
+    quality = _extractContent(data, start: offset, end: offset + 2);
+    offset += 2;
+
+    imageData = sbh.first.value.sublist(offset);
```

Comment:
Wouldn't it be easier to use `InputStream` here to parse the data, so you didn't have keep track of the offset?
It's just a mere suggestion I'll accept PR regardless.

Reply:
I didn't have time for better code. :(